firefox-ios/Client/Frontend/Settings/Main/Support/SendDataSetting.swift

```diff
@@ -84,29 +84,31 @@ class SendDataSetting: BoolSetting {
             attributedStatusText: statusText
         )
 
-        setupSettingDidChange()
+        setupSettingDidChange(for: sendDataType)
 
         // We make sure to set this on initialization, in case the setting is turned off
         // in which case, we would to make sure that users are opted out of experiments
         Experiments.setTelemetrySetting(prefs.boolForKey(prefKey) ?? true)
     }
 
-    private func setupSettingDidChange() {
+    private func setupSettingDidChange(for sendDataType: SendDataType) {
         self.settingDidChange = { [weak self] value in
-            // AdjustHelper.setEnabled($0)
-            DefaultGleanWrapper.shared.setUpload(isEnabled: value)
+            if sendDataType != .crashReports {
```

Comment:
As we add more logic for different sendDataType, it will become difficult to know what needs to be set for what type. I think we should move this logic outside the send data setting.

Reply:
I will take that in consideration when I will add set the next setting, in the next PR.